cypress/e2e/GUI/ui-funcionality-tests.cy.js

```diff
@@ -0,0 +1,155 @@
+describe('EngageSphere', () => {
+  beforeEach(() => {
+    Cypress.on('window:before:load', window => {
+      window.document.cookie = 'cookieConsent=accepted'
+    })
+    cy.visit('/');
+  });
+
+  it('Should maintain filters when returning from customer details', () => {
+    cy.get('[data-testid="size-filter"]')
+      .select('Small');
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+      cy.contains('button', 'Back')
+      .click();
```

Comment:
```suggestion
      .click();

```

É uma boa prática separar pré-condições, ações e resultados esperados por uma linha em branco.

Isso facilita identificar cada parte do teste.

Falaremos mais sobre isso no último encontro da _Test Design Masterclass_ quando abordarmos o padrão AAA (_Arrange, Act, Assert_).

Reply:
O mesmo vale para os demais testes.